### PR TITLE
Use createRequire with createRequireFromPath as fallback

### LINKED ISSUE #83: Use Module.createRequire instead of Module.createRequireFromPath
Currently getting this Node deprecation message:

```
(node:2968) [DEP0130] DeprecationWarning: Module.createRequireFromPath() is deprecated. Use Module.createRequire() instead.
```

### PR DESCRIPTION
**Pull request by ndresx**
`createRequireFromPath` is deprecated since Node `v12.2.0` ([source](https://nodejs.org/api/deprecations.html#deprecations_dep0130_module_createrequirefrompath)).

According to the [contributing documentation](https://github.com/facebookexperimental/rome/blob/master/.github/CONTRIBUTING.md#getting-started), everything above Node `v12` is fine, which means it would still need to support the previous implementation of `createRequireFromPath`.

Using `createRequire` was throwing a TS error, which why I have also updated the `@types/node` package. Therefore, the updated modules are now also part of this PR (separate commit via `npm install`), but I'm not sure how this process works here with committing them or not (same goes for the creation of the vendor file). Since it's a separate commit, it could be removed again, or the package update gets reverted _(would have to type `createRequire` a bit differently then)_.

Fixes #83 

----

PS: Could it be that there is a problem with the build pipeline step **"Run actions/setup-node@v1"**? https://github.com/facebookexperimental/rome/pull/90/checks?check_run_id=478564351

### CODE REVIEW

File: packages/@romejs/codec-websocket/index.ts [hunk 1]
```diff
@@ -240,7 +240,6 @@ export async function createClient(
 
     req.on('upgrade', (res, socket, head) => {
       if (res.headers['sec-websocket-accept'] !== digest) {
-        res.end();
```
Comment (ndresx) [inline packages/@romejs/codec-websocket/index.ts]: `end` doesn't exist on this response type (https://nodejs.org/api/http.html#http_class_http_incomingmessage). `socket.end()` should be enough.

**Comment by sebmck** [comment]
Thank you for doing this! I'll review when I get a chance.

> PS: Could it be that there is a problem with the build pipeline step **"Run actions/setup-node@v1"**? https://github.com/facebookexperimental/rome/pull/90/checks?check_run_id=478564351

Yeah it appears so. It was reported here actions/setup-node#116 but it's happening consistently now.

File: packages/@romejs/events/bridgeCreators.ts [hunk 2]
```diff
@@ -208,7 +208,7 @@ export function createBridgeFromChildProcess<B extends Bridge>(
   });
 
   proc.on('message', msg => {
-    bridge.handleMessage(msg);
+    bridge.handleMessage(msg as BridgeMessage);
```
Comment (sebmck) [inline packages/@romejs/events/bridgeCreators.ts]: Why was this change necessary?

File: packages/@romejs/node/index.ts [hunk 3]
```diff
@@ -18,7 +18,9 @@ function getRequire(folder: AbsoluteFilePath = CWD_PATH): RequireFunction {
     return cached;
   }
 
-  const req = mod.createRequireFromPath(folder.join());
+  const filename = folder.join();
+  const req =
+    mod.createRequire(filename) || mod.createRequireFromPath(filename);
```
Comment (sebmck) [inline packages/@romejs/node/index.ts]: Ditto here.

File: packages/@romejs/core/common/utils/executeMain.ts [hunk 4]
```diff
@@ -51,7 +51,9 @@ export default async function executeMain(
     setImmediate,
     setInterval,
     setTimeout,
-    require: internalModule.createRequireFromPath(filename),
+    require:
+      internalModule.createRequire(filename) ||
+      internalModule.createRequireFromPath(filename),
```
Comment (sebmck) [inline packages/@romejs/core/common/utils/executeMain.ts]: I think what you meant was

```suggestion
      internalModule.createRequire === undefined ? internalModule.createRequireFromPath(filename) : internalModule.createRequire(filename),
      
```

Comment (ndresx) [reply packages/@romejs/core/common/utils/executeMain.ts]: Ups, yes indeed! 😓 I've changed it, thanks for finding this.

File: packages/@romejs/events/bridgeCreators.ts (hunk 2 above)
Comment (ndresx) [reply packages/@romejs/events/bridgeCreators.ts]: With the update of `@types/node` to its latest version, `msg` is now from type `Serializable` and not `any` anymore https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/child_process.d.ts#L60.

> Argument of type 'Serializable' is not assignable to parameter of type 'BridgeMessage'.
  Type 'string' is not assignable to type 'BridgeMessage'

Alternatively, I could change the argument type in `handleMessage` to make it more transparent, or revert the update of the types module if it's too much action overall for this.

**Comment by sebmck** [comment]
Looks good to me, thank you! Thanks for updating the definitions too!
